# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/fetch/FetchStreamReader.h
# Commit: 682576598147
# Full Hash: 68257659814773dd5ce01d8dc9962eb3cd2313c9
# Author: Andrew Sutherland <asutherland@asutherland.org>
# Date: 2022-08-03 03:42:48
# Regressor Bug: 1770630
# File Overlap Count: 1
# Description:
#   Bug 1770630 - Worker stream readers should contribute to busy count. r=dom-worker-reviewers,jstutte
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D149185
# ==============================================================================

diff -r 08d3ed24aab9 -r 682576598147 dom/fetch/FetchStreamReader.h
--- a/dom/fetch/FetchStreamReader.h	Wed Aug 03 00:26:29 2022 +0000
+++ b/dom/fetch/FetchStreamReader.h	Wed Aug 03 00:36:01 2022 +0000
@@ -19,7 +19,7 @@
 
 class ReadableStream;
 class ReadableStreamDefaultReader;
-class WeakWorkerRef;
+class StrongWorkerRef;
 
 class FetchStreamReader final : public nsIOutputStreamCallback {
  public:
@@ -66,7 +66,8 @@
 
   nsCOMPtr<nsIAsyncOutputStream> mPipeOut;
 
-  RefPtr<WeakWorkerRef> mWorkerRef;
+  RefPtr<StrongWorkerRef> mWorkerRef;
+  RefPtr<StrongWorkerRef> mAsyncWaitWorkerRef;
 
   RefPtr<ReadableStreamDefaultReader> mReader;
 